pkg/operation/botanist/waiter.go

```diff
@@ -336,3 +336,20 @@ func (b *Botanist) WaitForControllersToBeActive() error {
 		return true, false, nil
 	})
 }
+
+// WaitUntilNodesDeleted waits until no nodes exist in the shoot cluster anymore.
+func (b *Botanist) WaitUntilNodesDeleted() error {
+	return wait.PollImmediate(5*time.Second, 600*time.Second, func() (bool, error) {
```

Comment:
Please `context`ify using our `retry` methods that keep track of the last error.

Reply:
Where is the context coming from? We don't have a context yet, or?

Reply:
The context is being passed in via the flow (see https://github.com/gardener/gardener/blob/master/pkg/utils/flow/taskfn.go#L24). If a superior function calls this function, then still its `context` should be used with proper child contexts (`context.WithTimeout`).